### PR TITLE
adjusts color in membership table to invert if value is negative

### PR DESCRIPTION
**Pull request by adambray**
e.g. tardiness

### CODE REVIEW

File: app/helpers/application_helper.rb [hunk 1]
```diff
@@ -25,29 +25,38 @@ def average_status collection
...
+    # if the input is a 'bad' value, flip and scale the value accordingly.
+    # e.g. a good value would be %HW complete, and 75% would be green
+    #
+    # but a bad value might be "% absent", and 10% would be yellow
+    input = (100 - (input* 3)) if type == :bad
+
     case input
-    when 0...25
-      return "s0"
+    when -1000...25
```
Comment (jshawl) [inline app/helpers/application_helper.rb]: is it sometimes negative? is this magic?

File: app/helpers/application_helper.rb [hunk 2]
```diff
@@ -25,29 +25,38 @@ def average_status collection
...
     when 25...50
-      return "s1"
+      return "status_bad"
     when 50...75
-      return "s2"
-    when 75..100
-      return "s3"
+      return "status_mediocre"
+    when 75..1000
+      return "status_good"
```
Comment (jshawl) [inline app/helpers/application_helper.rb]: printing CSS classes from a ruby method feels like a code smell. why the change from integers? also maybe a data-attribute is the best fit here:

```html
<p data-status="<%= @status %>>"</p>
```

File: app/helpers/application_helper.rb (hunk 1 above)
Comment (adambray) [inline app/helpers/application_helper.rb]: Agree that the numbers should be extracted into named constants, good catch.

As for the values, they can be negative in the case where the input is of type `:bad`. See the formula and how it subtracts a multiple of input from 100. So a 'bad' input of 100 would be (100 - (3*100) == -200.

This sort of formula is necessary to scale bad inputs appropriately. Probably an opportunity for refactoring here though.

File: app/helpers/application_helper.rb (hunk 2 above)
Comment (adambray) [inline app/helpers/application_helper.rb]: Printing CSS classes using ERB is a pretty common pattern that I've seen... and in this case, this change isn't introducing that... it was already printing CSS classes... this PR just renames the classes to be more clear.

Using `data-status` would require JS just to apply the styles, which would introduce additional complexity and indirection.

Comment (adambray) [inline app/helpers/application_helper.rb]: To be clear, these weren't integers before, they were CSS classes, eg `"s0"`.

**Comment by adambray** [comment]
@jshawl followup thoughts?

**Comment by jshawl** [comment]
:+1:
